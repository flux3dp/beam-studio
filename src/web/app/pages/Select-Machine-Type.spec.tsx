import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const mockWrite = jest.fn();
jest.mock('app/actions/beambox/beambox-preference', () => ({
  write: mockWrite,
}));

jest.mock('helpers/i18n', () => ({
  lang: {
    initialize: {
      no_machine: 'I don\'t have a machine now.',
      select_machine_type: 'Select Your Model',
    },
  },
}));

// eslint-disable-next-line import/first
import selectMachineType from './Select-Machine-Type';

describe('test Select-Machine-Type', () => {
  test('should render correctly', () => {
    const SelectMachineType = selectMachineType();
    const wrapper = mount(<SelectMachineType />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('.btn-action').at(0).simulate('click');
    expect(mockWrite).toHaveBeenNthCalledWith(1, 'model', 'fbm1');
    expect(mockWrite).toHaveBeenNthCalledWith(2, 'workarea', 'fbm1');
    expect(window.location.hash).toBe('#initialize/connect/select-connection-type');

    wrapper.find('.btn-action').at(1).simulate('click');
    expect(mockWrite).toHaveBeenNthCalledWith(3, 'model', 'fbb1b');
    expect(mockWrite).toHaveBeenNthCalledWith(4, 'workarea', 'fbb1b');
    expect(window.location.hash).toBe('#initialize/connect/select-connection-type');

    wrapper.find('.btn-action').at(2).simulate('click');
    expect(mockWrite).toHaveBeenNthCalledWith(5, 'model', 'fbb1p');
    expect(mockWrite).toHaveBeenNthCalledWith(6, 'workarea', 'fbb1p');
    expect(window.location.hash).toBe('#initialize/connect/select-connection-type');

    wrapper.find('.btn-link').simulate('click');
    expect(mockWrite).toHaveBeenCalledTimes(6);
    expect(window.location.hash).toBe('#initialize/connect/skip-connect-machine');
  });
});
