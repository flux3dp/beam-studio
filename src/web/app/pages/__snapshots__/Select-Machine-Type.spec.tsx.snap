// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Select-Machine-Type should render correctly 1`] = `
<SelectMachineType>
  <View
    className={
      Object {
        "initialization": true,
        "modal-window": true,
      }
    }
    content={
      <div
        className="connect-machine"
      >
        <div
          className="top-bar"
        />
        <div
          className="select-machine-type"
        >
          <h1
            className="main-title"
          >
            Select Your Model
          </h1>
          <div
            className="btn-h-group"
          >
            <button
              className="btn btn-action"
              onClick={[Function]}
              type="button"
            >
              beamo
            </button>
            <button
              className="btn btn-action"
              onClick={[Function]}
              type="button"
            >
              Beambox
            </button>
            <button
              className="btn btn-action"
              onClick={[Function]}
              type="button"
            >
              Beambox Pro
            </button>
          </div>
          <div
            className="btn btn-link"
            onClick={[Function]}
          >
            I don't have a machine now.
          </div>
        </div>
      </div>
    }
    disabledEscapeOnBackground={false}
    onClose={[Function]}
    onOpen={[Function]}
  >
    <div
      className="initialization modal-window"
    >
      <div
        className="modal-background"
        onClick={[Function]}
      />
      <div
        className="modal-body"
      >
        <div
          className="connect-machine"
        >
          <div
            className="top-bar"
          />
          <div
            className="select-machine-type"
          >
            <h1
              className="main-title"
            >
              Select Your Model
            </h1>
            <div
              className="btn-h-group"
            >
              <button
                className="btn btn-action"
                onClick={[Function]}
                type="button"
              >
                beamo
              </button>
              <button
                className="btn btn-action"
                onClick={[Function]}
                type="button"
              >
                Beambox
              </button>
              <button
                className="btn btn-action"
                onClick={[Function]}
                type="button"
              >
                Beambox Pro
              </button>
            </div>
            <div
              className="btn btn-link"
              onClick={[Function]}
            >
              I don't have a machine now.
            </div>
          </div>
        </div>
      </div>
    </div>
  </View>
</SelectMachineType>
`;
